gapis/resolve/memory.go

```diff
@@ -195,38 +196,49 @@ func expandCharArrays(ctx context.Context, typedRanges []*service.TypedMemoryRan
 		}
 
 		if slice := memType.GetSlice(); slice != nil && slice.GetUnderlying() == types.CharType {
-			typedRange.Value.GetSlice().Representation = &pod.Value{
-				Val: &pod.Value_String_{
-					String_: valAsString(typedRange.Value.GetSlice().GetValues()),
-				},
-			}
+			representation, values := valAsString(typedRange.Value.GetSlice().GetValues())
+			typedRange.Value.GetSlice().Representation = representation
+			typedRange.Value.GetSlice().Values = values
 		} else if arr := memType.GetArray(); arr != nil && arr.GetElementType() == types.CharType {
-			typedRange.Value.GetArray().Representation = &pod.Value{
-				Val: &pod.Value_String_{
-					String_: valAsString(typedRange.Value.GetArray().GetEntries()),
-				},
-			}
+			representation, entries := valAsString(typedRange.Value.GetArray().GetEntries())
+			typedRange.Value.GetArray().Representation = representation
+			typedRange.Value.GetArray().Entries = entries
 		}
 	}
 	return typedRanges, nil
 }
 
-// valAsString combines the values in the array into one string enclosed by quotation marks (")
-func valAsString(values []*memory_box.Value) string {
+// valAsString converts the int values to char and generates a string that omits the
+// null terminator and is enclosed by quotation marks (")
+func valAsString(values []*memory_box.Value) (*pod.Value, []*memory_box.Value) {
```

Comment:
I think this function should be split. 

Now it does two different things.

One returns a value as pod.Value(not as string anymore)
The other one, modifies the argument.

Reply:
Done